정유경/D2_1940.java

```diff
@@ -0,0 +1,35 @@
+package D2;
+
+import java.util.Scanner;
+
+public class D2_1940 {
+	public static void main(String[] args) {
+		Scanner sc = new Scanner(System.in);
+		int T = sc.nextInt();
+		
+		for(int k=1; k<=T; k++) {
+			int N = sc.nextInt();
+			int speed = 0; //초기 속도
+			int dist = 0; //초기 이동거리
+			
+			for(int i=0; i<N; i++) {
+				int a = sc.nextInt();
+				
+				if(a == 0) {
+					dist += speed;
+				} else if(a == 1) {
+					int b = sc.nextInt();
+					speed += b;
+					dist += speed;
+				} else if(a == 2) {
+					int b = sc.nextInt();
+					speed -= b;
+					if(speed < 0)
+						speed = 0;
+					dist += speed;
+				}
+			}
```

Comment:
```suggestion
			for(int i=0; i<N; i++) {
				int a = sc.nextInt();
				
                               if(a == 1)speed += sc.nextInt();
				if(a == 2) speed -= sc.nextInt();
				if(speed < 0) speed = 0;
				
				dist += speed;
			}
```
이부분 논리적으로는 문제 없지만, 더 클린하게 짤 수 있다고 생각해서 남겨요 !! 

1. if문 내에서 공통적으로 `dist += speed` 부분이 반복 되므로, 이 부분만 마지막에 빼면 코드를 효율적으로 작성할 수 있습니다.
2. else if로 조건문 파이프라인을 만들기보다는, 논리적으로 동일하다면 if문으로 별개의 조건으로 빼는 것이 더 명확합니다. (만약 else if 문이 길어지는 경우에는 switch문을 쓰는것이 더 명확합니다.)
3. 중첩 분기문 또한 논리적으로 동일하다면 별도로 분리하는 것이 명확하여 이해하기 쉽습니다.

Reply:
좋은 피드백 감사합니다!^-^